app/src/main/java/ch/epfl/skysync/viewmodel/InFlightViewModel.kt

```diff
@@ -0,0 +1,506 @@
+          factory =
+              object : ViewModelProvider.Factory {
+                override fun <T : ViewModel> create(modelClass: Class<T>): T {
+                  return InFlightViewModel(repository) as T
+                }
+              })
+    }
+  }
+
+  private val locationTable = repository.locationTable
+  private val flightTable = repository.flightTable
+  private val flightTraceTable = repository.flightTraceTable
+
+  /** If the view model has been initialized */
+  private var initialized = false
+
+  /**
+   * The user ID, initialized in the graph (and not at creation in MainActivity as we don't have it
+   * then)
+   */
+  private var _userId: String? = null
+
+  /** List of listeners listening to the location of users in the ongoing flight */
+  private val locationListeners = mutableListOf<ListenerRegistration>()
+
+  /** List of listeners listening to confirmed flights scheduled today where the user is assigned */
+  private val flightListeners = mutableListOf<ListenerRegistration>()
+
+  /** Map User ID -> User of the users in the ongoing flight */
+  private val users = mutableMapOf<String, User>()
+
+  /** ID of the pilot of the current flight */
+  private var pilotId: String? = null
+
+  /** The job that runs the timer */
+  private var timerJob: Job? = null
+
+  /** The timestamp of the start of the flight */
+  private var startTimestamp = 0L
+
+  private val _loading = MutableStateFlow(true)
+
+  /** If the view model is loading something */
+  val loading = _loading.asStateFlow()
+
+  private val _counter = MutableStateFlow(0L)
+
+  /** The current counter value in milliseconds.* */
+  val rawCounter = _counter.asStateFlow()
+
+  /** The current value of the counter formatted as a string in the format "HH:MM:SS". */
+  val counter =
+      _counter
+          .map { DateUtility.formatTime(it) }
+          .stateIn(
+              viewModelScope, started = WhileUiSubscribed, initialValue = DateUtility.formatTime(0))
+
+  private val _flightStage = MutableStateFlow(FlightStage.IDLE)
+
+  /** The current flight stage */
+  val flightStage: StateFlow<FlightStage> = _flightStage.asStateFlow()
+
+  private val _currentLocations = MutableStateFlow<Map<String, Pair<User, Location>>>(emptyMap())
+
+  /** Map User ID -> (User, Location) of the current location of each user in the ongoing flight */
+  val currentLocations: StateFlow<Map<String, Pair<User, Location>>> =
+      _currentLocations.asStateFlow()
+
+  private val _flightLocations = MutableStateFlow<List<Location>>(emptyList())
+
+  /** List of the successive locations through which the balloon (pilot) has passed */
+  val flightLocations: StateFlow<List<Location>> = _flightLocations.asStateFlow()
+
+  /** List of confirmed flights scheduled today where the user is assigned */
+  private val _confirmedFlights: MutableStateFlow<List<ConfirmedFlight>> =
+      MutableStateFlow(emptyList())
+
+  val startableFlights: StateFlow<List<ConfirmedFlight>> =

```

Comment:
Each day a user can be assigned to max 2 flights. Instead of showing both flights maybe we want to show only the current one i.e. in the morning we show the morning flight and in the evening the evening flight (if the user is assigned to a flight)

Reply:
Make sense, gone for it